Jenkinsfile

```diff
@@ -212,6 +218,42 @@ pipeline {
                 }
             }
         }
+
+        stage('7. Push Docker Images To GoogleCloud') {
+            when {
+                expression {
+                    return isReleaseBranch() || params.PUSH_TO_GOOGLE_CLOUD
+                }
+            }
+            agent {
+                docker {
+                    registryCredentialsId HARBOR_CREDS
+                    registryUrl HARBOR_URL
+                    image JAVA_IMAGE
+                    label NODE_LABEL
+                }
+            }
+            steps {
+                timeout(time: 30, unit: 'MINUTES') {
+                    script {
+                        def gitCommitHash = env.GIT_COMMIT
+                        def imageTags = "${versionText},${gitCommitHash}"
+                        withDockerCredentials(DOCKERHUB_CREDS, "FROM_") {
+                            withGCRCredentials(VORVAN_CRED) {
+                                def gcrCreds = readFile("${GCR_JSON_KEY}")
+                                withEnv(['TO_DOCKER_USERNAME=_json_key', "TO_DOCKER_PASSWORD=${gcrCreds}"]) {
```

Comment:
tested this a couple of times, if inlined with gradle command the file content will show up on jenkins console logs (not ideal)

Reply:
maybe something like this? https://github.com/h2oai/mlops-byom-images/blob/main/Jenkinsfile#L176-L193

Reply:
and have 2 separate stages?

Reply:
@nkpng2k we do unfortunately, because docker fetches the base img usually form dockerhub or just checks its signature, we actually had issues with pushing to gcr when dockerhub was not authenticated...